pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -88,6 +94,7 @@ public SegmentDeletionManager(String dataDir, HelixAdmin helixAdmin, String heli
     _helixClusterName = helixClusterName;
     _propertyStore = propertyStore;
     _defaultDeletedSegmentsRetentionMs = TimeUnit.DAYS.toMillis(deletedSegmentsRetentionInDays);
+    _segmentDeletionListeners = new ConcurrentLinkedQueue<>();
```

Comment:
I have the same comment as Jackie. I am not sure why there is a concurrent queue here

Reply:
I initially though since anyone can call register() dynamically (before / after start), it would make sense to ensure concurrent adds are supported. But now all listener instances are created before start(). LMK if this approach looks good